server/apps/main/templates/main/moderate_experience.html

```diff
@@ -306,13 +306,49 @@ <h5>Moderation reasons to be passed back to the author</h5>
         </template>
 
         <div class='form-group'>
-          <input type="submit" class="btn btn-outline-dark btn-lg float-right" value="Submit">
+          <input type="submit" class="btn btn-outline-dark btn-lg float-right" id="submitBtn" value="Submit">
         </div>
       </form>
     </div>
 
     <script src="{% static 'js/moderation-replies.js' %}"></script>
   </div>
 </section>
+
+  <template id="template-text-not-reviewed-comments" class="template-hidden">
+    <p>It's not possible to record <i>Moderation reasons</i> for stories that haven't been reviewed.</p>
+    <p>Please set the <i>Moderation status</i> to <i>in review</i> to hold reasons for later use, or set the status to <i>accepted</i> or <i>rejected</i> to send the reasons to the author now.</p>
+  </template>
+
+  <template id="template-text-reject-no-comments" class="template-hidden">
+    <p>You are rejecting the story without providing a reason. Are you sure you want to continue?</p>
+  </template>
+
+  <template id="template-text-approve-comments" class="template-hidden">
+    <p>You have included rejection moderation reasons, but are still aproving the story. Are you sure you want to continue?</p>
+  </template>
```

Comment:
I guess this is more a question of what we want: Should users be able to approve w/ rejection reasons given? 

I guess we also allow flagging triggering sections for which we apply the trigger label, right? In that case approving while labeling parts is expected behaviour?

Reply:
Yes, that was my thinking too. It could potentially make sense to flag issues, even for an approved story, in case the moderator is forcing a change to the triggering labels.

We could add a condition on this (check whether the labels have changed), or even just skip the check/dialogue in the case of an approval?

Reply:
It might be a bit overkill to check for it through the JS client-side validation, but maybe what we should validate on the server side is that _approved_ moderated experiences can only contain moderation feedback that's trigger warning-related and not _ground for rejection_-related?

Reply:
Let me give this a go. Technically I think it'd be easier to catch "red+approve" submissions on the client side, because the moderation types come from the HTML template and the code doesn't actually know anything about them. It wouldn't be sensible to split the reasons across the code and templates, but we could send the type (red/amber) back as part of the POST submit. I'd also prefer not to have the check done just on the client side.

Reply:
Yeah, I wonder if we should move the checking into the DB backend somewhere which we could then also load on the client side? That way we could validate on both depending on the need. But that's probably it's own PR as it would be quite a big change! 

So I'd be happy to have a minimal adjustment here and make an improvement issue instead of squeezing it in here 🙂

Reply:
I've updated it so it only allows amber replies through on approval. There's a check at both ends (client submit and server form validation); it's not an ideal approach, but avoids more significant changes.

Reply:
I only tried the check through JS in the frontend but that works really well. And the code for the backend looks great too!